uint8[] data
uint32 sample_rate
string format
uint32 sample_width
